### PR TITLE
sorter: remove file sorter

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- File sorter has not been maintained and tested for a long time, and no one we know of uses it in production.

### What is changed and how it works?
- Removed file sorter.
- For compatibility of running changefeeds, we use Unified Sorter in place when the user has previously set `sort-engine` to `file`.
- The CLI will refuse the set `sort-engine` to `file`.

### Check List 

Tests 

 - Integration test

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

- Remove file sorter.

### CODE REVIEW

File: cdc/processor/pipeline/sorter.go [hunk 1]
```diff
@@ -68,22 +66,11 @@ func (n *sorterNode) Init(ctx pipeline.NodeContext) error {
...
-				err = os.MkdirAll(sortDir, 0o755)
-				if err != nil {
-					return errors.Annotate(cerror.WrapError(cerror.ErrProcessorSortDir, err), "create dir")
-				}
-			} else {
-				return errors.Annotate(cerror.WrapError(cerror.ErrProcessorSortDir, err), "sort dir check")
-			}
+	case model.SortUnified, model.SortInFile /* `file` becomes an alias of `unified` for backward compatibility */ :
+		if sortEngine == model.SortInFile {
+			log.Warn("File sorter is obsolete. Please revise your changefeed settings and use unified sorter",
```
Comment (overvenus) [inline cdc/processor/pipeline/sorter.go:71]:
```suggestion
			log.Warn("File sorter is obsolete and replaced by unified sorter. Please revise your changefeed settings",
```

**Review by overvenus — COMMENTED** [review]
Rest LGMT

File: cmd/client_changefeed.go [hunk 2]
```diff
@@ -336,10 +336,13 @@ func verifyChangefeedParameters(ctx context.Context, cmd *cobra.Command, isCreat
 		}
 	}
 	switch sortEngine {
-	case model.SortUnified, model.SortInMemory, model.SortInFile:
+	case model.SortUnified, model.SortInMemory:
+	case model.SortInFile:
+		// obsolete. But we keep silent here. We create a Unified Sorter when the owner/processor sees this option
+		// for backward-compatibility.
 	default:
 		return nil, errors.Errorf("Creating changefeed with an invalid sort engine(%s), "+
```
Comment (ben1009) [inline cmd/client_changefeed.go:344]: the `+` here is just for more readable ?

Comment (liuzix) [reply cmd/client_changefeed.go:344]: Yes

**Review by amyangfei — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**